packages/plugin-credentials/src/credentials.js

```diff
@@ -246,6 +274,80 @@ const Credentials = SparkPlugin.extend({
     return Reflect.apply(SparkPlugin.prototype.set, this, [attrs, options]);
   },
 
+  /**
+   * Converts a CI SAML Bearer Token to an OAuth Bearer Token.
+   * @param {Object} options
+   * @param {Object} options.scope
+   * @param {Object} samlData Response body from the CI SAML endpoint.
+   * @private
+   * @return {Promise} Resolves with the bot's credentials.
+   */
+   @oneFlight
+  _getOauthBearerToken(options, samlData) {
+    this.logger.info(`credentials: exchanging SAML Bearer Token for OAuth Bearer Token`);
+
+    const vars = {
+      client_id: `CLIENT_ID`,
+      client_secret: `CLIENT_SECRET`
+    };
+
+    for (const key in vars) {
+      if (!has(this.config, key)) {
+        const baseVar = vars[key];
+        return Promise.reject(new Error(`config.credentials.${key} or CISCOSPARK_${baseVar} or COMMON_IDENTITY_${baseVar} or ${baseVar} must be defined`));
+      }
+    }
+
+    return this.spark.request({
+      method: `POST`,
+      uri: this.config.tokenUrl,
+      form: {
+        /* eslint camelcase: [0] */
+        grant_type: `urn:ietf:params:oauth:grant-type:saml2-bearer`,
+        assertion: samlData.BearerToken,
+        scope: options.scope
+      },
+      auth: {
+        user: this.config.client_id,
+        pass: this.config.client_secret,
+        sendImmediately: true
+      },
+      shouldRefreshAccessToken: false
+    })
+      .then((res) => new Token(res.body, {parent: this}));
+  },
+
+  /**
+   * Retrieves a CI SAML Bearer Token
+   * @private
+   * @return {Promise} Resolves with an Object containing a `BearerToken` and an
+   * `AccountExpires`
+   */
+   @oneFlight
+  _getSamlBearerToken(options) {
+    this.logger.info(`credentials: requesting SAML Bearer Token`);
+
+    if (!options.orgId) {
+      return Promise.reject(new Error(`\`options.orgId\` is required`));
+    }
+
+    if (!options.name) {
+      return Promise.reject(new Error(`\`options.name\` is required`));
+    }
+
+    if (!options.password) {
+      return Promise.reject(new Error(`\`options.password\` is required`));
+    }
+
+    return this.spark.request({
+      method: `POST`,
+      uri: `{this.config.samlUrl}/${options.orgId}/v2/actions/GetBearerToken/invoke`,
```

Comment:
Seems like this is missing a `$` at the front. I don't think the string interpolation will turn that into a variable without it.